### ISSUE TITLE
[SDL3]SDL_SetWindowFullscreen makes the window fullscreen on the wrong monitor

### ISSUE THREAD

**Issue by edo9300** [libsdl-org/SDL#11571]
On Windows, if "Show window contents while dragging" is disabled, moving a non maximized window from one screen to another and snapping it to maximized, will make a subsequent call to `SDL_SetWindowFullscreen` maximize it to the monitor it was previously in, instead of the current one, this works properly on SDL2

**Comment by Kontrabant** [comment]
Just to check, are you setting a video mode with `SDL_SetWindowFullscreenMode()` before entering fullscreen, or is this fullscreen desktop?

**Comment by edo9300** [comment]
No, it's default windowed borderless fullscreen, I'm not switching to exclusive fullscreen. If the show window contents is is enabled this issue doesn't arise instead

**Comment by Kontrabant** [comment]
Found the issue, and working on a fix now.

**Comment by edo9300** [comment]
I retested this interaction on the latest sdl3 release, and it seems to be wrong again (it was working on sdl 3.1.8), but already in sdl 3.2.0 it starts behaving with this broken behaviour again

**Comment by slouken** [comment]
@Kontrabant, can you take a look?

**Comment by Kontrabant** [comment]
Will do.

**Comment by Kontrabant** [comment]
After dragging, the maximized window is made fullscreen on the correct display, but it is put on the wrong display due to the fix needed for #12006. Working on it now.

**Comment by Kontrabant** [comment]
The regression should be fixed now, without regressing #12006 either.

**Comment by edo9300** [comment]
Can confirm it works properly on the current master branch
